### ISSUE TITLE
Google Chrome 75.0.3770.80 HEADLESS no longer works inside Docker container

### ISSUE THREAD

**Issue by mramato** [karma-runner/karma-chrome-launcher#198]
This was originally reported as https://github.com/karma-runner/karma/issues/3328 but the Karma team asked it be reported here instead.

In short, Chrome 74 headless worked flawless with karma-chrome-launcher from inside a Docker container but when Chrome 75 releases karma can no longer connect.  Several folks with otherwise valid configuration starting hitting this bug shortly after Chrome 75 was released.

For completeness, here is the launcher configuration being used (but as I said, the only thing that changed between working/not working was Chrome version

```js
        customLaunchers: {
            ChromeDocker: {
                base: 'ChromeHeadless',
                flags: [
                    '--no-sandbox'
                ]
            }
        },
```

**Comment by rogeriopvl** [comment]
Thanks for reporting @mramato 

I’ll have a look at it as soon as possible.

**Comment by FnTm** [comment]
@rogeriopvl hate to poll you, but have you had a chance to look at this? This issue has rendered parts of our CI effectively dead.

**Comment by rcollette** [comment]
Hmm, I was going to check because I regularly build my Angular apps in Docker and I just noticed that even though I have this to install Chromium on alpine

```dockerfile
FROM node:10.15.3-alpine as base
MAINTAINER richard.collette@pb.com
RUN apk add --update --no-cache git bash curl zip libexif udev chromium chromium-chromedriver xvfb ttf-freefont

ENV CHROME_BIN chromium-browser
```
It is only installing 
chromium (72.0.3626.121-r0)

While https://pkgs.alpinelinux.org/packages?name=chromium&branch=v3.10 shows the latest release as being 73.0.3683.103-r0

What is the base image and chromium install that is being discussed here?   I might be able to modify mine quickly and verify.

**Comment by zps938** [comment]
Using the --disable-dev-shm-usage flag fixed this for me.

**Comment by luninroman** [comment]
> Using the `--disable-dev-shm-usage` flag fixed this for me.

Did the trick for us as well, thanks!

**Comment by mramato** [comment]
Thanks @zps938 that did the trick!.  I also found this puppeteer issue where they discuss it and decide to make it a default option for them because it impacted so many users: https://github.com/GoogleChrome/puppeteer/issues/1834

@rogeriopvl do you agree that adding `--disable-dev-shm-usage` as one of the default headless args is a good idea?

**Comment by rogeriopvl** [comment]
@mramato yes, at first glance I see no problem in adding that as default param. Will have a closer look at it later to day to make sure there are no implications.

**Comment by charleszardo** [comment]
Using `--disable-dev-shm-usage` also fixed the problem for me.  Thanks @zps938

**Comment by sarala75** [comment]
Tests in docker container using Google headless chrome are failing with latest Chrome release (76)on 7/30/19. I was using CHROMEDRIVER_VERSION 74.0.3729.6 with last latest version of Chrome (as was available prior to 7/30). Please advice

**Comment by rogeriopvl** [comment]
@sarala75 I suspect that's a different issue

@mramato I included the `--disable-dev-shm-usage` in the default headless options. Should be out when v3.1.0 is published (PR #209).

**Comment by rubiesonthesky** [comment]
Flag `--disable-dev-shm-usage` may break CircleCI runs. Is there a way to disable it? Also if its needed only in Chrome > 75 can it be default only for using that?

Ref. https://github.com/ember-cli/ember-cli/issues/8536

**Comment by rogeriopvl** [comment]
Thanks for the heads up @rubiesonthesky , I'm going to have a look at it.
